### DISCUSSION TITLE
Allow Grafana to proxy datasources for another Grafana instance

### DISCUSSION THREAD

**Discussion by nberkley** [grafana/grafana#45808] [Data Sources]
We're in the process of moving our Grafana instance and much of our monitoring infrastructure to the cloud, but we have several datasources we use today that we don't want to directly expose on the public internet in order for them to be reachable. We could stand up proxies for each type of datasource individually, but that will quickly become burdensome, and it turns out we already have a piece of software that knows how to proxy requests to all of these datasources: Grafana.

The idea would be to support defining datasources to be proxied by another Grafana instance (with a matching datasource config) rather than itself. In this way an on-prem Grafana instance could be configured to talk to the datasources we need just as it is today, while the cloud hosted instance could then have corresponding datasources configured to use the on-prem instance as proxy.

This would allow us to keep all our dashboards and alerts together in one Grafana instance in the cloud, while still supporting on-prem datasources, and without introducing a mountain of different proxies for various technologies.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
